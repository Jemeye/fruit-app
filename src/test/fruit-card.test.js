import React from 'react';
import {render, screen} from '@testing-library/react';
import FruitCard from '../components/fruit-card';
describe('Fruit Card', () => {

    it('should render correctly', () => {
        const fruit = {
            id: 1,
            name: 'Apple',
            family: 'Rosaceae',
            order: 'Rosids',
            genus: 'Malus',
            nutritions: {
                calories: 52,
                fat: 0.1,
                protein: 0.3,
                sugar: 10,
                carbohydrates: 10,
            },
        };
        const { asFragment } = render(<FruitCard fruit={fruit} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should display the correct information of a fruit', () => {
        const fruit = {
            id: 1,
            name: 'Apple',
            family: 'Rosaceae',
            order: 'Rosids',
            genus: 'Malus',
            nutritions: {
                calories: 52,
                fat: 0.1,
                protein: 0.3,
                sugar: 10,
                carbohydrates: 10,
            },
        };
        render(<FruitCard fruit={fruit} />);
        
        // Checking if all elements are displayed on the page   
    expect(screen.queryByText(/Apple/i)).not.toBeNull();
    expect(screen.queryByText(/Rosaceae/i)).not.toBeNull();
    });
});