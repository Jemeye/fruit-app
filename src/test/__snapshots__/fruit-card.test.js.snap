// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fruit Card should render correctly 1`] = `
<DocumentFragment>
  <div
    class="fruit-card"
  >
    <div
      class="fruit-image"
    >
      <img
        alt="Fruit"
        src="   /Apple.jpg"
      />
    </div>
    <div
      class="fruit-description"
    >
      <h2>
        Apple
      </h2>
      <p>
        This fruit belongs to the family Rosaceae.
      </p>
      <button>
        View 
      </button>
    </div>
  </div>
</DocumentFragment>
`;
